langchain4j-mongodb-atlas/src/main/java/dev/langchain4j/store/embedding/mongodb/MappingUtils.java

```diff
@@ -0,0 +1,76 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import org.bson.Document;
+
+import java.util.Set;
+
+class MappingUtils {
+
+    private MappingUtils() throws InstantiationException {
+        throw new InstantiationException("can't instantiate this class");
+    }
+
+    static MongoDBDocument toMongoDbDocument(String id, Embedding embedding, TextSegment textSegment) {
+        if (textSegment == null) {
+            return new MongoDBDocument(id, embedding.vectorAsList(), null, null);
+        }
+        return new MongoDBDocument(id, embedding.vectorAsList(), textSegment.text(), textSegment.metadata().asMap());
+    }
+
+    static EmbeddingMatch<TextSegment> toEmbeddingMatch(MongoDBMatchedDocument matchedDocument) {
+        TextSegment textSegment = null;
+        if (matchedDocument.getMetadata() != null) {
+            textSegment = TextSegment.from(matchedDocument.getText(), Metadata.from(matchedDocument.getMetadata()));
+        }
+        return new EmbeddingMatch<>(matchedDocument.getScore(), matchedDocument.getId(), Embedding.from(matchedDocument.getEmbedding()), textSegment);
+    }
+
+    static Document toIndexMapping(IndexMapping indexMapping) {
```

Comment:
should this be called `fromIndexMapping`?

Reply:
done